### PR TITLE
BC-5063 - Add feature element menu

### PR DESCRIPTION
**Pull request by bischofmax**
# Short Description

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/components/feature-board-file-element/FileContentElement.vue [hunk 1]
```diff
@@ -23,13 +23,7 @@
...
 			@upload:file="onUploadFile"
 		/>
-		<ContentElementMenu
-			v-if="isEditMode"
-			v-bind="elementPositionProps"
-			@move-down:element="onMoveElementDown"
-			@move-up:element="onMoveElementUp"
-			@delete:element="onDeleteElement"
-		/>
+		<slot v-if="isEditMode" :elementName="fileProperties?.name"></slot>
```
Comment (SevenWaysDP) [inline src/components/feature-board-file-element/FileContentElement.vue:26]: please create named slot name="menu"

File: src/components/feature-board-file-element/upload/FileUpload.vue [hunk 2]
```diff
@@ -1,19 +1,23 @@
 <template>
-	<div>
-		<div v-if="!fileWasPicked">
+	<v-app-bar flat color="transparent">
```
Comment (bischofmax) [inline src/components/feature-board-file-element/upload/FileUpload.vue:2]: dense

Comment (SevenWaysDP) [inline src/components/feature-board-file-element/upload/FileUpload.vue:2]: image

toolbar: dense
FileInput: hide-hints and remove padding left
3DotMenu button anywhere  smaller ?

File: src/components/feature-board-file-element/upload/FileUpload.vue [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+		<div v-if="!fileWasPicked" class="w-100">
```
Comment (SevenWaysDP) [inline src/components/feature-board-file-element/upload/FileUpload.vue:3]: why we need w-100 ?

File: src/components/feature-board-file-element/upload/FileUpload.vue [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
-				indeterminate
-			/>
-		</div>
-	</div>
+
+		<v-progress-linear
+			v-else
+			data-testid="board-file-element-progress-bar"
+			indeterminate
+			class="w-100"
```
Comment (SevenWaysDP) [inline src/components/feature-board-file-element/upload/FileUpload.vue:14]: why we need w-100 ?

File: src/components/feature-board-file-element/content/display/image-display/ImageDisplay.vue [hunk 5]
```diff
@@ -6,6 +6,10 @@
 			:src="previewUrl"
 			:alt="alternativeText"
 		/>
+		<v-app-bar flat color="transparent" class="menu w-100">
```
Comment (SevenWaysDP) [inline src/components/feature-board-file-element/content/display/image-display/ImageDisplay.vue:9]: why we need w-100 ?

File: src/components/feature-board-file-element/upload/FileUpload.vue (hunk 3 above)
Comment (SevenWaysDP) [reply src/components/feature-board-file-element/upload/FileUpload.vue:3]: why v-if="!fileWasPicked" not on file picker

File: src/components/feature-board-content-element-menu/ContentElementMenu.vue [hunk 6]
```diff
@@ -0,0 +1,97 @@
+<template>
+	<BoardMenu scope="element">
+		<BoardMenuAction
+			v-if="hasMultipleElements && !isFirstElement"
+			data-testid="board-file-element-edit-menu-move-up"
```
Comment (bergatco) [inline src/components/feature-board-content-element-menu/ContentElementMenu.vue:5]:
```suggestion
			data-testid="content-element-menu-move-up"
```

File: src/components/feature-board-content-element-menu/ContentElementMenu.vue [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+			<template #icon>
+				<VIcon>
+					{{ mdiArrowCollapseUp }}
+				</VIcon>
+			</template>
+			{{ $t("components.board.action.moveUp") }}
+		</BoardMenuAction>
+		<BoardMenuAction
+			v-if="hasMultipleElements && !isLastElement"
+			data-testid="board-file-element-edit-menu-move-down"
```
Comment (bergatco) [inline src/components/feature-board-content-element-menu/ContentElementMenu.vue:17]:
```suggestion
			data-testid="content-element-menu-move-down"
```

File: src/components/feature-board-file-element/content/display/file-description/FileDescription.vue [hunk 8]
```diff
@@ -1,11 +1,7 @@
 <template>
 	<div class="grey lighten-4">
-		<v-card-title
-			v-if="showTitle"
-			data-testid="board-file-element-display"
-			inactive
-		>
-			<div class="d-flex flex-no-wrap align-center text-truncate">
+		<v-app-bar v-if="showTitle" flat color="transparent">
+			<v-toolbar-title data-testid="board-file-element-display" inactive>
```
Comment (bergatco) [inline src/components/feature-board-file-element/content/display/file-description/FileDescription.vue:4]:
```suggestion
			<v-toolbar-title data-testid=file-description-toolbar" inactive>
```

File: src/components/feature-board-file-element/content/display/file-description/FileDescription.vue [hunk 9]
```diff
@@ -15,12 +11,16 @@
 				>
 
 				<span
-					:class="spanClasses"
+					class="subtitle-1 font-weight-bold"
 					data-testid="board-file-element-display-file-name"
```
Comment (bergatco) [inline src/components/feature-board-file-element/content/display/file-description/FileDescription.vue:15]:
```suggestion
					data-testid="file-description-name"
```

**Review by SevenWaysDP — APPROVED** [review]

**Pull request merged**